cmd/multus-daemon/main.go

```diff
@@ -111,39 +130,51 @@ func main() {
 		}
 		logging.Verbosef("Generated MultusCNI config: %s", generatedMultusConfig)
 
-		if err := configManager.PersistMultusConfig(generatedMultusConfig); err != nil {
+		multusConfigFile, err = configManager.PersistMultusConfig(generatedMultusConfig)
+		if err != nil {
 			_ = logging.Errorf("failed to persist the multus configuration: %v", err)
 		}
 
-		go func(stopChannel chan<- struct{}, doneChannel chan<- struct{}) {
-			if err := configManager.MonitorPluginConfiguration(configWatcherStopChannel, doneChannel); err != nil {
+		go func(ctx context.Context, doneChannel chan<- struct{}) {
+			if err := configManager.MonitorPluginConfiguration(ctx, doneChannel); err != nil {
 				_ = logging.Errorf("error watching file: %v", err)
 			}
-		}(configWatcherStopChannel, configWatcherDoneChannel)
-
-		<-configWatcherDoneChannel
+		}(ctx, configWatcherDoneChannel)
 	} else {
 		if err := copyUserProvidedConfig(multusConf.MultusConfigFile, multusConf.CniConfigDir); err != nil {
 			logging.Errorf("failed to copy the user provided configuration %s: %v", multusConf.MultusConfigFile, err)
 		}
 	}
 
-	serverDone := false
-	configWatcherDone := false
-	for {
-		select {
-		case <-configWatcherDoneChannel:
-			logging.Verbosef("ConfigWatcher done")
-			configWatcherDone = true
-		case <-serverDoneChannel:
-			logging.Verbosef("multus-server done.")
-			serverDone = true
+	signalCh := make(chan os.Signal, 16)
+	signal.Notify(signalCh, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		for sig := range signalCh {
+			logging.Verbosef("caught %v, stopping...", sig)
+			cancel()
 		}
+	}()
 
-		if serverDone && configWatcherDone {
-			return
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		<-configWatcherDoneChannel
+		logging.Verbosef("ConfigWatcher done")
+		if multusConf.MultusConfigFile == "auto" && multusConfigFile != "" {
+			logging.Verbosef("Delete old config")
```

Comment:
nit: could we have a `logging.Verbosef("Deleting old config @ %v",multusConfigFile)`

Reply:
fixed.